enf/resource_enf_network.go

```diff
@@ -0,0 +1,95 @@
+package enf
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xaptum/go-enf/enf"
+)
+
+func resourceEnfNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceEnfNetworkCreate,
+		Read:   resourceEnfNetworkRead,
+		Update: resourceEnfNetworkUpdate,
+		Delete: resourceEnfFirewallRuleDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceEnfNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	domainID := meta.(*EnfClient).DomainID
+
+	params := &enf.NetworkRequest{
+		Name:        enf.String(d.Get("name").(string)),
+		Description: enf.String(d.Get("description").(string)),
+	}
+
+	client := meta.(*EnfClient).Client
+	network, _, err := client.Network.CreateNetwork(context.Background(), strconv.FormatInt(domainID, 10), params)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*network.Network)
+
+	return resourceEnfNetworkRead(d, meta)
+}
+
+func resourceEnfNetworkRead(d *schema.ResourceData, meta interface{}) error {
+	address := d.Id() // this is the network address
+
+	client := meta.(*EnfClient).Client
+	network, resp, err := client.Network.GetNetwork(context.Background(), address)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("name", network.Name)
+	d.Set("description", network.Description)
+	d.Set("status", network.Status)
+
+	return nil
+}
+
+func resourceEnfNetworkUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*EnfClient).Client
+
+	if d.HasChange("name") || d.HasChange("description") {
```

Comment:
Is there any need to check for `hasChange` on name and description?  Those are the only two mutable fields, so if this method is called, doesn't that imply that one of them has changed?  Genuine questions.

Reply:
My guess (the docs aren't very explicit) is that the `hasChange` function really only matters for partial state, a concept that we don't employ at all. I removed the if block completely and everything still works.